sympy/core/tests/test_assumptions.py

```diff
@@ -27,7 +27,7 @@ def test_zero():
     assert z.is_complex is True
     assert z.is_noninteger is False
     assert z.is_irrational is False
-    assert z.is_imaginary is False
+    assert z.is_imaginary is True
```

Comment:
Doesn't work on `master` as of now. A bug.

Reply:
It's not a bug but rather a different definition. Currently imaginary is defined as meaning imaginary and not zero so `x.is_imaginary == x.is_complex and re(x).is_zero and im(e).is_nonzero`. This PR attempts to change it so that the definition is `x.is_imaginary == x.is_complex and re(x).is_zero` which is cleaner and parallels `is_real`. This is equivalent except for whether or not zero is considered to be imaginary: on master it is not but this PR changes that.